src/doc/src/guide/project-layout.md

```diff
@@ -48,6 +48,8 @@ files, place a `main.rs` file along with the extra [*modules*][def-module]
 within a subdirectory of the `src/bin`, `examples`, `benches`, or `tests`
 directory. The name of the executable will be the directory name.
 
+Binaries, examples, benches and integration tests follow `kebab-case` naming style. Modules within those targets are `snake_case` following the [Rust standard](https://rust-lang.github.io/rfcs/0430-finalizing-naming-conventions.html).
```

Comment:
```suggestion
By convention, binaries, examples, benches and integration tests follow `kebab-case` naming style, unless there are compatibility reasons to do otherwise (e.g. compatibility with a pre-existing binary name). Modules within those targets are `snake_case` following the [Rust standard](https://rust-lang.github.io/rfcs/0430-finalizing-naming-conventions.html).
```

Reply:
Could you also include the changes from @joshtriplett ?  The current text is very prescriptive, leaving out room for people to make decisions to follow other conventions when there is a motivating case.

Reply:
Sure. Made the modification.